packages/flutter_tools/lib/src/build_info.dart

```diff
@@ -113,3 +115,23 @@ HostPlatform getCurrentHostPlatform() {
 
   return HostPlatform.linux_x64;
 }
+
+/// Returns the top-level build output directory.
+String getBuildDirectory() {
+  return config.getValue('build-dir') ?? 'build';
```

Comment:
Looks like we'll choke in a few places if this is specified as an absolute path.  Do we want to detect that here and throw?